openmetadata-ui/src/main/resources/ui/src/components/Entity/EntityLineage/AppPipelineModel/AddPipeLineModal.tsx

```diff
@@ -108,9 +108,16 @@ const AddPipeLineModal = ({
     return;
   }, [selectedEdge, edgeSearchValue]);
 
+  const debounceOnSearch = useCallback(debounce(getSearchResults, 300), []);
+
+  const handleChange = (value: string): void => {
+    setEdgeSearchValue(value);
+    debounceOnSearch(value);
+  };
+
   useEffect(() => {
     getSearchResults(edgeSearchValue);
-  }, [edgeSearchValue]);
+  }, []);
```

Comment:
```suggestion
  }, [edgeSearchValue]);
```

Reply:
Explicitly kept it like this.
We want to call getSearchResults on launch with empty string.
Post that all the calls are made on debounce.